lib/ble/ble.toit

```diff
@@ -713,6 +727,69 @@ class DataBlock:
     result.replace 2 data
     return result
 
+  hash-code -> int:
+    return type.hash-code * 31 + data.hash-code
+
+  operator== other/any:
+    if other is not DataBlock: return false
+    other-block := other as DataBlock
+    return type == other-block.type and data == other-block.data
+
+  stringify --add-prefix/bool=true -> string:
+    prefix := add-prefix ? "DataBlock: " : ""
+    if type == TYPE-RAW: return "$(prefix)raw=$data"
+    if type == TYPE-FLAGS:
+      flags-string := ""
+      if (flags & BLE-ADVERTISE-FLAGS-LIMITED-DISCOVERY) != 0:
+        flags-string += "LE Limited Discoverable Mode, "
+      if (flags & BLE-ADVERTISE-FLAGS-GENERAL-DISCOVERY) != 0:
+        flags-string += "LE General Discoverable Mode, "
+      if (flags & BLE-ADVERTISE-FLAGS-BREDR-UNSUPPORTED) != 0:
+        flags-string += "BR/EDR Not Supported, "
+      if (flags & BLE-ADVERTISE-FLAGS-LE-BREDR-CONTROLLER) != 0:
+        flags-string += "LE and BR/EDR Controller, "
+      if (flags & BLE-ADVERTISE-FLAGS-LE-BREDR-HOST) != 0:
+        flags-string += "LE and BR/EDR Host, "
+      if flags-string.is-empty: return "$(prefix)flags=$(%x flags)"
+      flags-string = flags-string[.. flags-string.size - 2]
+      return "$(prefix)flags=$(%x flags) ($flags-string)"
+    if type == TYPE-SERVICE-UUIDS-16-INCOMPLETE:
+      return "$(prefix)services-16-incomplete=$services-16"
+    if type == TYPE-SERVICE-UUIDS-16-COMPLETE:
+      return "$(prefix)services-16-complete=$services-16"
+    if type == TYPE-SERVICE-UUIDS-32-INCOMPLETE:
+      return "$(prefix)services-32-incomplete=$services-32"
+    if type == TYPE-SERVICE-UUIDS-32-COMPLETE:
+      return "$(prefix)services-32-complete=$services-32"
+    if type == TYPE-SERVICE-UUIDS-128-INCOMPLETE:
+      return "$(prefix)services-128-incomplete=$services-128"
+    if type == TYPE-SERVICE-UUIDS-128-COMPLETE:
+      return "$(prefix)services-128-complete=$services-128"
+    if type == TYPE-NAME-SHORTENED:
+      return "$(prefix)name-shortened=$name"
+    if type == TYPE-NAME-COMPLETE:
+      return "$(prefix)name-complete=$name"
+    if type == TYPE-TX-POWER-LEVEL:
+      return "$(prefix)tx-power-level=$tx-power-level"
+    if type == TYPE-SERVICE-DATA-16 or type == TYPE-SERVICE-DATA-32 or type == TYPE-SERVICE-DATA-128:
+      bit-size/int := ?
+      if type == TYPE-SERVICE-DATA-16: bit-size = 16
+      else if type == TYPE-SERVICE-DATA-32: bit-size = 32
+      else if type == TYPE-SERVICE-DATA-128: bit-size = 128
+      else: unreachable
+      data-string := ""
```

Comment:
I think I'd build this up as a list instead and then join.

Reply:
done.